### PR TITLE
Turbo version cleanup test

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

File: Common/cpp/NativeReanimatedModule.cpp [hunk 1]
```diff
@@ -206,6 +206,51 @@ void NativeReanimatedModule::onEvent(std::string eventName, std::string eventAsS
...
+  //  2 - worklets
+  //  3 - appliers
+  jsi::Function fun = value.getObject(rt).asFunction(rt);
+  std::shared_ptr<jsi::Function> funPtr(new jsi::Function(std::move(fun)));
+
+  scheduler->scheduleOnUI([&rt, funPtr, this, option]() {
+    std::string ids;
+    switch(option) {
+      case 1: {
+        for(auto it = sharedValueRegistry->sharedValueMap.begin(); it != sharedValueRegistry->sharedValueMap.end(); ++it) {
```
Comment (Szymon20000) [inline Common/cpp/NativeReanimatedModule.cpp]: Can you use foreach loop?

File: Common/cpp/headers/ApplierRegistry.h [hunk 2]
```diff
@@ -11,10 +11,10 @@
 #include <unordered_map>
 
 class ApplierRegistry {
-  std::unordered_map<int, std::shared_ptr<Applier>> renderAppliers;
-  std::map<std::string, std::unordered_map<int, std::shared_ptr<Applier>>> eventAppliers;
-  std::map<int, std::string> eventMapping;
   public:
+    std::unordered_map<int, std::shared_ptr<Applier>> renderAppliers;
```
Comment (Szymon20000) [inline Common/cpp/headers/ApplierRegistry.h]: Use getters/setters instead.

File: Example/testComponents/CleanUpTest.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import Animated, { useSharedValue, useWorklet, RegistersState } from 'react-native-reanimated';
+import { View, Text } from 'react-native';
+
+const workletBody = function(a, b, c) {
```
Comment (Szymon20000) [inline Example/testComponents/CleanUpTest.js]: If you don't create Worklet instance here then each Event Worklet will register separate Worklet. Is this really what you want?

File: src/reanimated2/RegistersState.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import NativeModule from './NativeReanimated';
+
+
+export default class RegistersState {
```
Comment (Szymon20000) [inline src/reanimated2/RegistersState.js]: Can you export instance instead of class?

File: Common/cpp/NativeReanimatedModule.cpp (hunk 1 above)
Comment (karol-bisztyga) [reply Common/cpp/NativeReanimatedModule.cpp]: fixed 70dd9c2b456dc5edb71581e9c70aded6a060f804

File: Common/cpp/headers/ApplierRegistry.h (hunk 2 above)
Comment (karol-bisztyga) [reply Common/cpp/headers/ApplierRegistry.h]: fixed 70dd9c2b456dc5edb71581e9c70aded6a060f804

File: Example/testComponents/CleanUpTest.js (hunk 3 above)
Comment (karol-bisztyga) [reply Example/testComponents/CleanUpTest.js]: yes this is it exactly

File: src/reanimated2/RegistersState.js (hunk 4 above)
Comment (karol-bisztyga) [reply src/reanimated2/RegistersState.js]: fixed d84f2ab0bb3275696ef078668869069e14f1f7b9

File: ios/native/NativeProxy.mm [hunk 5]
```diff
@@ -93,7 +93,7 @@ + (void*) getNativeReanimatedModule:(void*)jsInvokerVoidPtr
 + (NSArray<NSArray*>*)getChangedSharedValues
 {
   NSMutableArray *changed = [NSMutableArray new];
-  for(auto & sharedValue : nativeReanimatedModule->sharedValueRegistry->sharedValueMap) {
+  for(auto & sharedValue : nativeReanimatedModule->sharedValueRegistry->getSharedValueMap()) {
```
Comment (Szymon20000) [inline ios/native/NativeProxy.mm]: Shouldn't we make the same change in NativeProxy.cpp?

Comment (karol-bisztyga) [reply ios/native/NativeProxy.mm]: yes, fixed 688b7e4b5f7734a1a7a6bc47b735250fcd7614d1

**Pull request merged**